File: api/src/main/java/com/bindglam/neko/api/config/KeyConfigurable.java
package com.bindglam.neko.api.config;

import net.kyori.adventure.key.Key;
import net.kyori.adventure.key.KeyPattern;
import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.minimessage.MiniMessage;
import org.intellij.lang.annotations.Subst;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class KeyConfigurable implements Configurable<Key, String> {
    @Override
    public @Nullable Key load(@KeyPattern @Nullable String value) {
        if(value == null) return null;

        return Key.key(value);
    }
}

File: core/src/main/kotlin/com/bindglam/neko/config/KeyConfigurable.kt
package com.bindglam.neko.config

import com.bindglam.neko.api.config.Configurable
import net.kyori.adventure.key.Key
import org.bukkit.NamespacedKey

object KeyConfigurable : Configurable<NamespacedKey, String> {
    override fun load(value: String?): NamespacedKey? = value?.let { NamespacedKey.fromString(value) }
}